src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -66,6 +70,7 @@ public override int GetHashCode()
 
             public void Dispose()
             {
+                OwningDisposeScope?.WasDisposed(this);
```

Comment:
This should probably be guarded by the check for a null handle on line 86. Otherwise, a no-op dispose will increment the DisposeScope counter of disposed elements.

Reply:
makes sense